src/test/java/htsjdk/samtools/cram/structure/ContainerTest.java

```diff
@@ -1,149 +1,104 @@
 package htsjdk.samtools.cram.structure;
 
 import htsjdk.HtsjdkTest;
-import htsjdk.samtools.SAMRecord;
 import htsjdk.samtools.cram.CRAMException;
+import htsjdk.samtools.cram.ref.ReferenceContext;
+import htsjdk.samtools.cram.ref.ReferenceContextType;
 import org.testng.Assert;
 import org.testng.annotations.Test;
 
-public class ContainerTest extends HtsjdkTest {
+import java.util.Arrays;
 
+public class ContainerTest extends HtsjdkTest {
     @Test
     public static void singleRefSliceStateTest() {
-        final Slice slice = new Slice();
-        slice.sequenceId = 5;
+        final Slice slice = new Slice(new ReferenceContext(5));
         slice.alignmentStart = 10;
         slice.alignmentSpan = 15;
 
-        final Container container = new Container();
-        container.finalizeContainerState(slice);
+        final Container container = Container.initializeFromSlices(Arrays.asList(slice));
 
-        Assert.assertEquals(container.sequenceId, slice.sequenceId);
+        Assert.assertTrue(container.getReferenceContext().isMappedSingleRef());
+        Assert.assertEquals(container.getReferenceContext().getType(), ReferenceContextType.SINGLE_REFERENCE_TYPE);
+        Assert.assertEquals(container.getReferenceContext(), slice.getReferenceContext());
```

Comment:
Many of the blocks of asserts in these tests could be replaced with one of the `ContainerFactoryTest.assertContainerState` methods. Not sure if `SliceTests.assertSliceState` could be reuseable as well.